discovery/spec.md

```diff
@@ -917,108 +931,111 @@ Content-Type: application/json
+all mandatory attributes MUST be present in the request, and a Discovery
+Endpoint MUST reject requests that are missing such attributes.
+
+The Discovery Endpoint MUST ignore attempts to modify read-only or immutable
+attributes - such as `id` and `url`.
+
+Upon successfully processing the request, the HTTP response body MUST be
+the Service values resulting from the successful processing of the request.
+The response SHOULD include the current values for all attributes, even if
+the Discovery Endpoint added or modified some value during the processing
+of the request.
+
+The following responses are defined by this specification:
+- `200 OK` if specified Service was updated successfullly.
 - `400 Bad Request` if the `id` in the path and body are not the same or some
-  other constraint failure is found
-- `404 Not Found` if there is no Service with the specified `id`
-- `409 Conflict` if the given `epoch` was less than or equal to the existing
-  `epoch` of the Service with the given `id`
+  other constraint failure is found.
+- `409 Conflict` if the given `epoch` is not greater than the current value.
 
 Other responses are allowed, but not defined by this specification.
 
-The format of the HTTP Request MUST adhere to the following:
+The format of the HTTP request MUST adhere to the following:
 ```
 Content-Type: application/json
 
 {
   "url": "{url}",
-  "epoch": "{id}",
   "name": "{name}",
   ... remainder of Service attributes ...
 }
 ```
 
-The format of the `200 OK` HTTP Response MUST adhere to the following:
+The format of the `200 OK` HTTP response MUST adhere to the following:
 ```
 200 OK
 Content-Type: application/json
 
 {
   "id": "{id}",
-  "epoch": "{id}",
+  "epoch": "{epoch}",
   ... remainder of Service attributes ...
 }
 ```
 
-#### `DELETE /services/{id}`
+#### `DELETE /services/{id}[?epoch={epoch}]`
+
+This MUST delete the Service as identified by the `id` in the URL. If there
+is no Service with the specified `id` then it is considered to have already
+been deleted and the request will have no change to the Discovery Endpoint.
+A missing Service is not considered an error condition.
 
-This MUST delete the Service at the referenced URL.
+This specification does not define a Body for the request message and the
+presence of a Body (even with a Service definition, valid or not) MUST NOT
+result in an error.
 
-The request MAY include a body. Any such body MUST be an object containing an
-`id` matching the `{id}` in the path. That object MAY include an `epoch`
-value. If the Service corresponding with the specified `id` has a higher
-recorded `epoch` value than given, the request MUST fail.
+The request URL MAY include an OPTIONAL `epoch` query parameter, and if
+present and is not greater than the Service's current `epoch` value, then the
+request MUST fail.
 
-If other service attributes are included in the request, those MUST be
+If other service attributes are included in the request, those SHOULD be
```

Comment:
They would have to be as extra query parameters, right?

Reply:
Given [this](https://github.com/cloudevents/spec/pull/930/files/aee0e42152993400fa5e32af78d9d1256d958c95#r835604807), I believe so.  Query parameters are a part of the request but your statement is correct as written.  Are you suggesting that the language be more specific?

Reply:
It was more making sure I'd understood correctly. I'm not sure whether I'd agree with ignoring them at all... I tend to think that if a client puts a query parameter on a URL, they expect it to be used. But I suspect I'm alone in that :)

Reply:
hmm I think that sentence may have been left over from a previous version where the Body meant something. 
I'm going to remove it because we have this sentence:
```
Unknown query parameters on the APIs MUST be ignored.
```
in another part of the doc.